core/src/executor/evaluate/mod.rs

```diff
@@ -323,25 +325,28 @@ async fn evaluate_function<'a, 'b: 'a, 'c: 'a, T: GStore>(
                 let value = if fargs.len() == args.len() {
                     let mut hm = StdHashMap::new();
 
-                    args.iter().enumerate().try_for_each(|(i, arg)| -> Result<()> {
-                        arg.validate_type(&fargs[i].data_type)?;
-                        arg.validate_null(fargs[i].default.is_some())?;
-                        let value = if arg.is_null() {
-                            // fargs[i].default.as_ref().unwrap()
-                            &args[i]
-                        } else {
-                            &args[i]
-                        };
-                        hm.insert(fargs[i].name.to_owned(), value.to_owned());
-                        Ok(())
-                    })?;
+                    args.into_iter()
+                        .map(|expr| Value::try_from(expr).unwrap())
+                        .zip(fargs.iter())
+                        .try_for_each(|(arg, farg)| -> Result<()> {
+                            arg.validate_type(&farg.data_type)?;
+                            arg.validate_null(farg.default.is_some())?;
+                            let value = if arg.is_null() {
+                                // farg.default.as_ref().unwrap()
+                                &arg
+                            } else {
+                                &arg
+                            };
+                            hm.insert(farg.name.to_owned(), value.to_owned());
+                            Ok(())
+                        })?;
 
                     let row = Row::Map(hm);
                     let rowcontext = RowContext::new(name, Cow::Borrowed(&row), None);
                     let context = Rc::new(rowcontext);
```

Comment:
@panarch, i try to use context set variables in the body by their values passed in argument
i try to implement the context, i got this error
`returning this value requires that "'3" must outlive "'2"`